semgrep-core/src/parsing/tree_sitter/Parse_hcl_tree_sitter.ml

```diff
@@ -129,346 +123,408 @@ let rec map_anon_choice_get_attr_7bbf24f (env : env)
+        | `GTEQ tok -> (* ">=" *) (GtE, token env tok)
+        | `LT tok -> (* "<" *) (Lt, token env tok)
+        | `LTEQ tok -> (* "<=" *) (LtE, token env tok)
       in
       let v3 = map_expr_term env v3 in
-      todo env (v1, v2, v3)
+      (v1, v2, v3)
   | `Expr_term_choice_EQEQ_expr_term (v1, v2, v3) ->
       let v1 = map_expr_term env v1 in
       let v2 =
         match v2 with
-        | `EQEQ tok -> (* "==" *) token env tok
-        | `BANGEQ tok -> (* "!=" *) token env tok
+        | `EQEQ tok -> (* "==" *) (Eq, token env tok)
+        | `BANGEQ tok -> (* "!=" *) (NotEq, token env tok)
       in
       let v3 = map_expr_term env v3 in
-      todo env (v1, v2, v3)
+      (v1, v2, v3)
   | `Expr_term_choice_AMPAMP_expr_term (v1, v2, v3) ->
       let v1 = map_expr_term env v1 in
-      let v2 = match v2 with `AMPAMP tok -> (* "&&" *) token env tok in
+      let v2 = match v2 with `AMPAMP tok -> (* "&&" *) (And, token env tok) in
       let v3 = map_expr_term env v3 in
-      todo env (v1, v2, v3)
+      (v1, v2, v3)
   | `Expr_term_choice_BARBAR_expr_term (v1, v2, v3) ->
       let v1 = map_expr_term env v1 in
-      let v2 = match v2 with `BARBAR tok -> (* "||" *) token env tok in
+      let v2 = match v2 with `BARBAR tok -> (* "||" *) (Or, token env tok) in
       let v3 = map_expr_term env v3 in
-      todo env (v1, v2, v3)
+      (v1, v2, v3)
 
-and map_collection_value (env : env) (x : CST.collection_value) =
+and map_collection_value (env : env) (x : CST.collection_value) : expr =
   match x with
   | `Tuple (v1, v2, v3) ->
       let v1 = (* "[" *) token env v1 in
-      let v2 =
-        match v2 with Some x -> map_tuple_elems env x | None -> todo env ()
-      in
+      let v2 = match v2 with Some x -> map_tuple_elems env x | None -> [] in
       let v3 = (* "]" *) token env v3 in
-      todo env (v1, v2, v3)
+      G.Tuple (v1, v2, v3) |> G.e
   | `Obj x -> map_object_ env x
 
-and map_expr_term (env : env) (x : CST.expr_term) =
+and map_expr_term (env : env) (x : CST.expr_term) : expr =
   match x with
-  | `Lit_value x -> map_literal_value env x
+  | `Lit_value x -> L (map_literal_value env x) |> G.e
   | `Temp_expr x -> map_template_expr env x
   | `Coll_value x -> map_collection_value env x
-  | `Var_expr tok -> (* identifier *) token env tok
+  | `Var_expr tok ->
+      (* identifier *)
+      let id = str env tok in
+      N (H2.name_of_id id) |> G.e
   | `Func_call (v1, v2, v3, v4) ->
-      let v1 = (* identifier *) token env v1 in
+      let v1 = (* identifier *) str env v1 in
       let v2 = (* "(" *) token env v2 in
       let v3 =
-        match v3 with
-        | Some x -> map_function_arguments env x
-        | None -> todo env ()
+        match v3 with Some x -> map_function_arguments env x | None -> []
       in
       let v4 = (* ")" *) token env v4 in
-      todo env (v1, v2, v3, v4)
+      let n = N (H2.name_of_id v1) |> G.e in
+      Call (n, (v2, v3, v4)) |> G.e
```

Comment:
Please give these variables a name while it's fresh in your mind. We don't have time to reverse engineer your code.